dot/state/storage.go

```diff
@@ -305,3 +308,28 @@ func (s *StorageState) GenerateTrieProof(stateRoot common.Hash, keys [][]byte) (
 	encodedProofNodes [][]byte, err error) {
 	return proof.Generate(stateRoot[:], keys, s.db)
 }
+
+// GetRuntimeFromDB gets the runtime for the corresponding block hash from storageState
+func (s *StorageState) GetRuntimeFromDB(blockHash common.Hash) (instance runtime.Instance, err error) {
+	var stateRootHash *common.Hash
+	stateRootHash, err = s.GetStateRootFromBlock(&blockHash)
+	if err != nil {
+		return nil, fmt.Errorf("getting state root from block hash: %w", err)
+	}
```

Comment:
This should be deduped as a method on StorageState I think? 🤔  We have similar bodies in multiple places in this PR